cartridges/int_twilio/cartridge/controllers/Twilio.js

```diff
@@ -0,0 +1,85 @@
+'use strict';
+
+/**
+ * @namespace Twilio
+ */
+
+var server = require('server');
+
+var addToCOHelper = require('*/cartridge/scripts/helpers/addToCOHelper')
+// var utils = require('../scripts/utils/utils');
+
+/**
+* Product-Show : This endpoint is called to show the details of the selected product
+* @name Base/Twilio-SavePhoneNumber
+* @function
+* @memberof Twilio
+* @param {httpparameter} - csrf_token - a CSRF token
+ @param {middleware} - server.middleware.https
+* 
+*/
+server.post('SavePhoneNumber',
+    server.middleware.https,
+    function (req, res, next) {
+        var outOfStockForm = server.forms.getForm('outOfStockForm');
+        var Resource = require('dw/web/Resource');
+
+        if (outOfStockForm.valid) {
+            var productName = outOfStockForm.twilio.productName.htmlValue;
+            var phoneNumber = outOfStockForm.twilio.phone.htmlValue;
+            var productId = outOfStockForm.twilio.productId.htmlValue;
+
+            var addToCOResponse = addToCOHelper.addToCOHelper(productId, phoneNumber, productName);
+
+            if (addToCOResponse.existingPhoneNumber && addToCOResponse.success) {
+                res.json({
+                    success: true,
+                    msg: Resource.msg('success.message.existingPhoneNumber', 'forms', null)
+                });
+            } else if (!addToCOResponse.existingPhoneNumber && addToCOResponse.success) {
+                res.json({
+                    success: true,
+                    msg: Resource.msg('success.message.addToCO', 'forms', null)
+                });
+            } else if (!addToCOResponse.existingPhoneNumber && !addToCOResponse.success) {
+                res.json({
+                    success: false,
+                    msg: Resource.msg('error.twilio.message.parse.phone', 'forms', null)
+                });
+            }
+
+        } else {
+            res.json({
+                success: false,
+                msg: Resource.msg('error.twilio.message.parse.phone', 'forms', null)
+            });
+        }
```

Comment:
Very well organized controller. One note, do not use only forms.properties. It should be just for form definitions.

Reply:
fixed